yjit/src/virtualmem.rs

```diff
@@ -57,14 +57,39 @@ pub trait Allocator {
     fn mark_unused(&mut self, ptr: *const u8, size: u32) -> bool;
 }
 
-/// Pointer into a [VirtualMemory].
-/// We may later change this to wrap an u32.
-/// Note: there is no NULL constant for CodePtr. You should use Option<CodePtr> instead.
+/// Pointer into a [VirtualMemory] represented as an offset from the base.
+/// Note: there is no NULL constant for [CodePtr]. You should use `Option<CodePtr>` instead.
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Debug)]
 #[repr(C, packed)]
-pub struct CodePtr(NonNull<u8>);
+pub struct CodePtr(u32);
```

Comment:
I believe this would effectively revert https://github.com/ruby/ruby/pull/6792. is that intentional?

Reply:
It's fine, since the optimization from `NonNull` was only important when we stored `Option<CodePtr>` in a lot of places. We've stopped wrapping `CodePtr`s in the types we retain long term.

I was saw the size of `Block` and `Branch` go down before writing up the patch through rust-analyzer hover :)